ui/snippets/walletMenu/WalletTooltip.tsx

```diff
@@ -8,20 +9,31 @@ type Props = {
 };
 
 const WalletTooltip = ({ children, isDisabled, isMobile }: Props) => {
+  const router = useRouter();
   const [ isTooltipShown, setIsTooltipShown ] = useBoolean(false);
 
+  const { defaultLabel, label, localStorageKey } = React.useMemo(() => {
+    const isAppPage = router.pathname === '/apps/[id]';
+    const defaultLabel = <span>Your wallet is used to interact with<br/>apps and contracts in the explorer</span>;
+    const label = isAppPage ?
+      <span>Connect once to use your wallet with<br/>all apps in the DAppscout marketplace!</span> :
+      defaultLabel;
+    const localStorageKey = `${ isAppPage ? 'dapp-' : '' }wallet-connect-tooltip-shown-${ isMobile ? 'mobile' : 'desktop' }`;
```

Comment:
I wonder why there are separate keys for mobile and desktop devices. It seems unnecessary since user sessions and data cannot be shared between the two.

Reply:
The problem is that both components are rendered at the same time, and if you use 1 key, only the tooltip for one of the device types will be displayed. I tried using Show/Hide components, but the result was even worse, because the button itself appeared with a delay. If you have an idea, I'm ready to redo it, because I don't really like this solution myself

Reply:
Oh, I see now. Then, maybe the parent component should pass `isDisabled=true` prop if viewport size doesn't match. Will it work?